include/measurement_kit/http/http.hpp

```diff
@@ -58,10 +63,26 @@ ErrorOr<Url> parse_url_noexcept(std::string url);
 | | |  __/ (_| | |_| |  __/\__ \ |_
 |_|  \___|\__, |\__,_|\___||___/\__|
              |_|
+
+    HTTP request and response structs, logic to make requests.
 */
 
 using Headers = std::map<std::string, std::string>;
 
+class Request {
+  public:
+    std::string method;
+    Url url;
+    std::string protocol;
+    Headers headers;
+    std::string path;
+    std::string body;
+
+    Request() {}
+    Error init(Settings, Headers, std::string);
+    void serialize(net::Buffer &);
+};
```

Comment:
This class used to be called `RequestSerialized`. Now it's `Request`. I've made it public because I plan soon to chain Requests and Responses, which is useful *ex post* to inspect what really happened when you are redirected in the `web_connectivity` test. (cc: @hellais)